bentoml/_internal/bento/bento.py

```diff
@@ -334,6 +340,21 @@ def save(
             fs.mirror.mirror(self._fs, out_fs, copy_if_newer=False)
             self._fs.close()
             self.__fs = out_fs
+            event_properties = {
```

Comment:
should we use an attrs class to represent each event type's properties?  we already have cattrs for JSON serialization, and it provides type checking and validation for us

Reply:
We can